File: src/lib/api/scripture.ts
import { command } from "./client";
import type { DetectionResult } from "../../types";

/**
 * Deterministic content detection over a transcript (or any text).
 *
 * The parse itself happens in Rust — nothing about Scripture parsing is
 * reimplemented in the frontend.
 */
export function detectScripture(transcript: string): Promise<DetectionResult[]> {
  return command<DetectionResult[]>("detect_scripture", { transcript });
}

File: src-tauri/src/commands/scripture.rs
//! Scripture commands: deterministic detection over text.

use tauri::State;

use crate::errors::CommandResult;
use crate::models::content::{ContentDetector, DetectionResult};
use crate::state::AppState;

#[tauri::command]
pub fn detect_scripture(
    transcript: String,
    _state: State<'_, AppState>,
) -> CommandResult<Vec<DetectionResult>> {
    Ok(ContentDetector.detect(&transcript))
}
